### DISCUSSION TITLE
RMF fleet adapters <> RMF core

### DISCUSSION THREAD

**Discussion by Achllle** [osrf/rmf_core#311] [Q&A]
I'm wondering why the RMF fleet adapters are so tightly integrated with RMF core as opposed to communicating with it through ROS2 messages. It seems like because of the direct import (either python or c++), the adapters are required to live on the same compute device as RMF core. Is that statement correct?

If this is the case, all fleet adapters would be living on the same machine and if for some reason that machine goes down, fleets are unable to communicate with each other directly.

**Comment by gbiggs** [comment]
Thanks for your question. However we ask that you please ask questions on https://github.com/open-rmf/rmf/discussions following our new support guidelines: https://github.com/open-rmf/rmf/wiki/Support-guidelines
